src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs

```diff
@@ -85,7 +85,7 @@ public async Task SynchronizeWithServer()
             {
                 var bindingConfiguration = configurationProvider.GetConfiguration();
 
-                if (IsStandalone(bindingConfiguration) || !IsConnected() || !IsFeatureSupported())
+                if (IsStandalone(bindingConfiguration) || !IsConnected(out var serverInfo) || !IsFeatureSupported(serverInfo))
```

Comment:
This is a bit weak and in fact the problem can still occur: for example, if we pass the serverInfo checks and then fail when we actually try to call the server. 
But I think this is still better and safer, instead of the ugly null exception.